src/Components/Web/src/FileDownloader.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components;
+
+/// <summary>
+/// Provides an abstraction for downloading a file from a Blazor app.
+/// </summary>
+public sealed class FileDownloader : IFileDownloader
```

Comment:
Given we have a public interface, we should try to make this `internal` so we don't expose this as part of our public API.

Reply:
Making this class `internal` causes issues with registering the service on the WebAssembly side, as `FileDownloader` cannot be accessed by `WebAssemblyHostBuilder.cs`. In order to solve this, I believe I'd have to implement `/Services/WebAssemblyFileDownloader.cs`, which I think I mentioned on Monday. Are there any other solutions so that the `internal` class can be accessed in WebAssembly/WebView?

Reply:
> In order to solve this, I believe I'd have to implement `/Services/WebAssemblyFileDownloader.cs`

This is something we probably want to avoid as the actual implementation of the two classes would be the same. Instead, we should try to "share" the class amongst WebAssembly/Server/WebView.

I'd look into how we currently have [`PullFromJSDataStream`[(https://github.com/dotnet/aspnetcore/blob/main/src/Components/Shared/src/PullFromJSDataStream.cs) in a shared directory and then add it as a shared asset to be compiled in the `.csproj`:

https://github.com/dotnet/aspnetcore/blob/50d1f7b1fba231adbb97f674def6c5e3f4478b7b/src/Components/WebAssembly/WebAssembly/src/Microsoft.AspNetCore.Components.WebAssembly.csproj#L32

Reply:
Sounds good, thank you for the explanation!